test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -894,39 +894,108 @@ public async Task VerifyRestorePackageByRestoreNuGetPackagesContextMenu(ProjectT
-            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ClassLibrary, ProjectTargetFramework.V48, "TestProject");
+            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, "TestProject");
             VisualStudio.ClearOutputWindow();
             solutionService.SaveAll();
 
             CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
             NuGetUIProjectTestExtension uiwindow = nugetTestService.GetUIWindowfromProject(project);
             uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
 
-            var packageFolderPath = _pathContext.PackagesV2;
-            var installedPackageFolderPath = Path.Combine(packageFolderPath, "Contoso.A.1.0.0");
+            if (projectTemplate.Equals(ProjectTemplate.NetCoreConsoleApp))
+            {
+                packageFolderPath = _pathContext.UserPackagesFolder;
+                installedPackageFolderPath = Path.Combine(packageFolderPath, TestPackageName);
+            }
+            else
+            {
+                packageFolderPath = _pathContext.PackagesV2;
+                installedPackageFolderPath = Path.Combine(packageFolderPath, "Contoso.A.1.0.0");
+            }
+
             Directory.Exists(installedPackageFolderPath).Should().BeTrue();
 
-            // Act
+            // Act (we doesn't edit the "nuget.config" file by setting the "automatic" to "true" since it is enabled by default)
             Directory.Delete(packageFolderPath, true);
             Directory.Exists(packageFolderPath).Should().BeFalse();
             solutionService.Build();
 
             // Assert
             Directory.Exists(installedPackageFolderPath).Should().BeTrue();
         }
+
+
+        [DataTestMethod]
+        [Timeout(DefaultTimeout)]
+        public void VerifyPackageNotRestoredAfterDisablingPackageRestore()
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+
+            SolutionService solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.IOSLibraryApp, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            _pathContext.Settings.DisableAutoRestore();
+            CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
+            VisualStudio.ObjectModel.Shell.ToolWindows.ErrorHub.ShowErrors();
+
+            // Assert
+            VisualStudio.AssertErrorListContainsSpecificError("NuGet restore is currently disabled.");
+        }
+
+        [DataTestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task VerifyPackageNotRestoredAfterDisablingAutomaticInPackageRestoreSection()
```

Comment:
This one is still failing.

Reply:
The test has passed now, could you please help review these cases, thanks a lot!

Reply:
I don't see a successful run since Tuesday. Even if it is sometimes successful, if no changes were made then it's likely a flaky test. My first guess would be a race condition with completing the action and reading the result in the test.

> Expected boolean to be false, but found True.

Reply:
Hi @donnie-msft, thanks for your reminder, we updated the code yesterday and verified these cases more than ten times locally and pipeline, they all pass.